src/types/createRouteOptions.ts

```diff
@@ -50,28 +51,40 @@ export type WithoutParent = {
 }
 
 export type WithComponent<
-  TComponent extends Component | undefined = Component | undefined
+  TComponent extends Component = Component,
+  TParams extends Record<string, unknown> = Record<string, unknown>
 > = {
   /**
    * A Vue component, which can be either synchronous or asynchronous components.
    */
-  component?: TComponent,
+  component: TComponent,
```

Comment:
Adding the `WithoutComponents` overloads and making `WithComponent` and `WithComponents` have required properties fixed some issue with correctly identifying overloads. And means that the `isWithComponent` and `isWithComponents` utilities now just assert `is WithComponent(s)`